src/dynamo0p3.py

```diff
@@ -2619,11 +2886,24 @@ def generate(self):
         '''specifies which arguments appear in an argument list, their type
         and their ordering. Calls methods for each type of argument
         that can be specialised by a child class for its particular need'''
-        if self._kern.arguments.has_operator:
-            # operators require the cell index to be provided
+        if self._kern.arguments.has_operator():
+            # All operator types require the cell index to be provided
             self.cell_position()
-        # always pass the number of layers in the mesh
-        self.mesh_height()
+        # Pass the number of layers in the mesh unless this kernel is
+        # applying a CMA operator or doing a CMA matrix-matrix calculation
+        if self._kern.cma_operation not in ["apply", "matrix-matrix"]:
+            self.mesh_height()
+        # Pass the number of cells in the mesh if this kernel has a
+        # LMA operator argument
+        # TODO this code should replace the code that currently includes
```

Comment:
Another issue - we have a ticket (551) for it on SRS that I will migrate over to github.